crates/nu-engine/src/eval.rs

```diff
@@ -514,7 +564,15 @@ fn eval_element_with_input(
                         Some((*err_span, err_expr.clone(), *err_append_mode)),
                     );
 
-                    eval_call(engine_state, stack, &save_call, input).map(|_| {
+                    eval_call(
+                        engine_state,
+                        stack,
+                        &save_call,
+                        input,
+                        debug_context,
+                        debugger,
```

Comment:
what if the debugger/debug_context was part of engine_state?

Do we need to pull it out and make it separate?